src/operation/problem_results.jl

```diff
@@ -764,27 +706,28 @@ function read_expressions(
     len::Union{Int, Nothing} = nothing,
 )
     result_values =
-        read_expressions_with_keys(res, expressions; start_time = start_time, len = len)
+        read_results_with_keys(res, expressions; start_time = start_time, len = len)
     return Dict(encode_key_as_string(k) => v for (k, v) in result_values)
 end
 
 """
 Return the values for all expressions.
 """
 function read_expressions(res::IS.Results)
-    expressions = Dict(x => read_expression(res, x) for x in list_expression_names(res))
+    return Dict(x => read_expression(res, x) for x in list_expression_names(res))
 end
 
-function read_expressions_with_keys(
+function read_results_with_keys(
```

Comment:
Add note that the existence of this function might need to be reviewed

Reply:
I’m pretty sure the reason is that Clayton wanted to be able to treat problems and simulation problems the same in PowerGraphics.

Reply:
ok, lets leave it for now